libcaf_io/src/tcp.cpp

```diff
@@ -0,0 +1,76 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#include "caf/policy/tcp.hpp"
+
+#include "caf/logger.hpp"
+#include "caf/io/network/socket_utils.hpp"
+
+using caf::io::network::is_error;
+using caf::io::network::rw_state;
+using caf::io::network::native_socket;
+using caf::io::network::no_sigpipe_io_flag;
+
+namespace caf {
+namespace policy {
+
+rw_state tcp::read_some(size_t& result, native_socket fd, void* buf,
+                        size_t len) {
+  CAF_LOG_TRACE(CAF_ARG(fd) << CAF_ARG(len));
+  auto sres = ::recv(fd, reinterpret_cast<io::network::socket_recv_ptr>(buf),
```

Comment:
`recv` is used without including the right header (we shouldn't rely on other headers to include them for us).